File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    c_dict = {} 
    
    for _, kind in clothes:
        if kind not in c_dict:
            c_dict[kind] = 1
        else:
            c_dict[kind] += 1
    
    answer = 1
    # 옷 종류별로 옷 가짓수 + 1(안입기)을 계속 곱해줌
    for c in c_dict.values():
        answer *= (c+1)
    
    return answer - 1 # 전부 안입기인 경우 제외

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        Map<String, Integer> map = new HashMap<>();
        for (String[] cloth : clothes) {
            if (map.containsKey(cloth[1])) {
                map.put(cloth[1], map.get(cloth[1])+1);
            } else {
                map.put(cloth[1], 1);
            }
        }
        
        List<Integer> valueList = new ArrayList<>(map.values());
        for (int value : valueList) {
            answer *= (value + 1);
        }
        
        return answer - 1;
    }
}